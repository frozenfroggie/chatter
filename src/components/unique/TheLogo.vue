<template>
  <div class='logoContainer'>
    <div class='logo'>
      <div class='logoInner'>
      </div>
    </div>
    <div class='logoText'>
      <h1> CHATTER </h1>
    </div>
  </div>
</template>

<script>
export default {
}
</script>

<style scoped>
.logoContainer {
  grid-area: logoContainer;
  display: flex;
  transform: scale(0.5);
}
.logoText {
  color: rgba(250,250,250,0.9);
  padding-left: 20px;
}
.logo {
  width: 70px;
  height: 70px;
  border: 6px solid rgba(250,250,250,0.9);
  border-radius: 50%;
}
.logo:after {
  display: block;
  content: '';
  position: absolute;
  left: 51px;
  top: 51px;
  width: 20px;
  height: 20px;
  background-color: rgb(46, 59, 79);
  transform: skew(20deg,20deg);
}
.logo:before {
  display: block;
  content: '';
  position: absolute;
  left: 59px;
  top: 59px;
  width: 20px;
  height: 20px;
  background-color: rgba(250,250,250,0.9);
  transform: skew(20deg,20deg);
}
.logoInner {
  position: absolute;
  width: 34px;
  height: 34px;
  border: 5px solid rgba(250,250,250,0.9);
  border-radius: 50%;
  left: 20px;
  top: 18px;
  z-index: 1;
}
.logoInner:after {
  display: block;
  content: '';
  position: absolute;
  left: 1px;
  top: 18px;
  width: 14px;
  height: 14px;
  background-color: rgb(46, 59, 79);
  transform: skew(-20deg,-20deg);
}
.logoInner:before {
  display: block;
  content: '';
  position: absolute;
  left: -5px;
  top: 25px;
  width: 13px;
  height: 13px;
  background-color: rgba(250,250,250,0.9);
  transform: skew(-20deg,-20deg);
}
</style>
