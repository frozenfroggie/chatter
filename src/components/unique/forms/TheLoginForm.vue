<template>
  <div class="inputContainer">
    <div class="username">
      <icon name="user" scale="1"></icon>
      <input
        type="text"
        name="username"
        placeholder="Username"
        aria-label="Username"
        v-model="username">
    </div>
    <div class="password">
      <icon name="unlock" scale="1"></icon>
      <input
        type="password"
        name="password"
        placeholder="Password"
        aria-label="Password"
        v-model="password">
    </div>
  </div>
</template>

<script>

export default {
  data () {
    return {
      username: '',
      password: ''
    }
  }
}
</script>

<style scoped>
.inputContainer {
  grid-area: loginInput;
  background-color: rgb(38, 49, 66);
}
.username {
  height: 50%;
  border-top: 1px solid rgba(250,250,250,0.5);
}
.password {
  height: 50%;
  border-top: 1px solid rgba(250,250,250,0.5);
  border-bottom: 1px solid rgba(250,250,250,0.5);
}
input {
  height: 100%;
  width: 80%;
  padding-left: 10px;
  background-color: inherit;
  border: none;
  color: rgba(250,250,250,0.9);
}
@media only screen and (min-width: 768px) {
  .inputContainer {
    border-radius: 10px;
  }
  .username {
    border-radius: 10px 10px 0px 0px;
  }
  .password {
    border-radius: 0px 0px 10px 10px;
  }
}
</style>
