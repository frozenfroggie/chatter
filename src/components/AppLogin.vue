<template>
  <div class="login">

    <the-logo />

    <div class='loginHeader'> Login to your account </div>

    <the-login-form />

    <button type="button" name="" class='loginBtn' @click='$emit("login")'>LOG IN</button>

    <div class="forgotPassword"> Forgot password? </div>

    <div class='signup' @click='$router.push("/signup")'>
      Don't have an account? <span> Sign up </span>
    </div>

  </div>
</template>

<script>
import TheLogo from './unique/TheLogo.vue'
import TheLoginForm from './unique/forms/TheLoginForm.vue'

export default {
  components: {
    TheLogo,
    TheLoginForm
  }
}
</script>

<style scoped>
.login {
  position: absolute;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
  color: rgba(250,250,250,0.9);
  display: grid;
  grid-template-columns: 2fr 1fr;
  grid-template-rows: 1fr 120px 50px 50px 50px 80px 2fr 50px;
  grid-template-areas:
    ' . . '
    ' logoContainer logoContainer'
    ' loginHeader loginHeader '
    ' loginInput loginInput'
    ' loginInput loginInput'
    ' loginBtn forgotPassword'
    ' . . '
    ' signup signup ';
}
.loginHeader {
  grid-area: loginHeader;
  text-align: left;
  padding-left: 10%;
}
.loginBtn {
  grid-area: loginBtn;
  margin: 1.2em 1em;
  width: 80%;
  color: rgba(250,250,250,0.9);
  background-color: rgb(38, 49, 66);
  border: 1px solid rgba(250,250,250,0.5);
  border-radius: 22px;
}
.loginBtn:hover {
  cursor: pointer;
}
.forgotPassword {
  width: 100%;
  height: 100%;
  grid-area: forgotPassword;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  text-align: left;
  font-size: .7em;
  color: rgba(250,250,250,0.5);
  text-decoration: underline;
}
.forgotPassword:hover {
  cursor: pointer;
}
.signup {
  grid-area: signup;
  font-size: .85em;
}
.signup > span {
  font-weight: 600;
}
.signup > span:hover {
  cursor: pointer;
}
@media only screen and (min-width: 768px) {
  .login {
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-template-rows: 1fr 120px 50px 50px 50px 80px 2fr 50px;
    grid-template-areas:
      ' . . . . '
      ' . logoContainer logoContainer .'
      ' . loginHeader loginHeader . '
      ' . loginInput loginInput . '
      ' . loginInput loginInput . '
      ' . loginBtn forgotPassword . '
      ' . . . . '
      ' . signup signup .';
  }
}
@media only screen and (min-width: 1024px) {
  .login {
    grid-template-columns: 30% 1fr 1fr 30%;
  }
}
@media only screen and (min-width: 1280px) {
  .login {
    grid-template-columns: 32% 1fr 1fr 32%;
  }
}
</style>
