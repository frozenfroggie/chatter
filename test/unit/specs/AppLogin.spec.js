import { shallowMount } from '@vue/test-utils'
import AppLogin from '@/components/AppLogin'
import TheLogo from '@/components/unique/TheLogo'
import TheLoginForm from '@/components/unique/forms/TheLoginForm'

describe('AppLogin', () => {
  it('renders TheLogo component', () => {
    const wrapper = shallowMount(AppLogin)
    const TheLogoComponent = wrapper.find(TheLogo)
    expect(TheLogoComponent.is(TheLogo)).toBe(true)
  })
  it('renders TheLoginForm component', () => {
    const wrapper = shallowMount(AppLogin)
    const TheLoginFormComponent = wrapper.find(TheLoginForm)
    expect(TheLoginFormComponent.is(TheLoginForm)).toBe(true)
  })
  it('matches snapshot', () => {
    const wrapper = shallowMount(AppLogin)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
