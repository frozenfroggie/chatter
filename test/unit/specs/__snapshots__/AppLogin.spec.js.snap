// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLogin matches snapshot 1`] = `
<div class="login">
    <!---->
    <div class="loginHeader"> Login to your account </div>
    <!---->
    <button type="button" name="" class="loginBtn">LOG IN</button>
    <div class="forgotPassword"> Forgot password? </div>
    <div class="signup">
        Don't have an account? <span> Sign up </span></div>
</div>
`;
